### PR TITLE
update chai

### PR DESCRIPTION
**Pull request by dtaller**
Updated chai

Some tests in test/TestAlgorithm seem to fail (produce wrong results). Needs to be investigated. Was an issue prior to this chai library update. Needs to be investigated

### CODE REVIEW

File: src/care/KeyValueSorter.h [hunk 1]
```diff
@@ -223,8 +223,6 @@ class KeyValueSorter<T, RAJADeviceExec> {
          setFromArray(len, arr);
       }
 
-#if defined(CARE_ENABLE_IMPLICIT_CONVERSIONS)
```
Comment (adayton1) [inline src/care/KeyValueSorter.h:226]: Why were these removed?

Comment (dtaller) [reply src/care/KeyValueSorter.h:226]: Needed to compile both with and without implicit conversions disabled. I tried both, and I needed this  when CARE_ENABLE_IMPLICIT_CONVERSIONS was OFF also

Comment (adayton1) [reply src/care/KeyValueSorter.h:226]: The actual issue is that implicit conversions need to be turned off in CHAI as well, and they haven't been. I've got a fix that will take care of this, so you can revert these changes.

Comment (dtaller) [reply src/care/KeyValueSorter.h]: Ok, sounds good. I reverted the changes. Just make sure that things work with implicit conversions both on and off in you pull request afterwards.

Comment (adayton1) [reply src/care/KeyValueSorter.h]: It works. I double checked.

**Review by adayton1 — APPROVED** [review]
LGTM
